### PR TITLE
Release binaries for mac os and windows

### PR DESCRIPTION
**Pull request by ebmifa**
## Description 
Building binaries for Ubuntu and Windows

## Test Plan 
https://github.com/MystenLabs/sui/actions/runs/5661640075
End Result: https://github.com/MystenLabs/sui/releases/tag/devnet-v1.6.0

### CODE REVIEW

**Review by after-ephemera — APPROVED** [review]
Do we know approximately how much this will cost us running on `windows-latest` and `macos-latest`? Last I remember they weren't cheap.

**Comment by ebmifa** [comment]
> Do we know approximately how much this will cost us running on `windows-latest` and `macos-latest`? Last I remember they weren't cheap.

This workflow should only run a few times a week, at most. So, not sure how much that would cost us more. @brad-mysten  do you know?

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -36,39 +36,90 @@ env:
...
+        uses: arduino/setup-protoc@v1
+        # this avoids rate-limiting
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Install postgres (Windows)
+        if: ${{ matrix.os == 'windows-ghcloud' && env.sui_tag != 'main'  }}
+        shell: bash
+        run: |
+          choco install postgresql12 --force --params '/Password:root'
```
Comment (allan-bailey) [inline .github/workflows/release.yml:71]: Nothing newer than Pg12?  We're using Pg 15 in prod.

**Review by allan-bailey — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 1 above)
Comment (ebmifa) [reply .github/workflows/release.yml:71]: I copied this code from another workflow. So 🤷🏼

Comment (ebmifa) [reply .github/workflows/release.yml:71]: See https://github.com/MystenLabs/sui/blob/main/.github/workflows/rust.yml#L156

**Review by allan-bailey — APPROVED** [review]

**Pull request merged**